zebra-state/src/lib.rs

```diff
@@ -58,12 +61,17 @@ pub enum Request {
     },
     /// Get the block that is the tip of the current chain
     GetTip,
+    /// Get the set of hashes for every block in the state service
+    GetKnownBlockHashes {
+        /// the range of heights to restrict the blocks too
+        range: Range<BlockHeight>,
+    },
```

Comment:
Range queries will be useful in the future, but for the chain-extension logic I think we don't want to query by height but by hash:

> 3. For each response, starting from the beginning of the list, prune any block hashes already included in the state, stopping at the first unknown hash to get resp1', ..., respF'. (These lists may be empty).

We only have hashes, not heights, so we need to be able to query whether a particular block hash is in the current best chain.

Reply:
I think we need to query by both hash and height:
* hashes: to "prune any block hashes already included in the state"
* heights: to "Query the current state to construct the sequence of hashes" (on the main chain)